packages/@glimmer/integration-tests/lib/suites/has-block.ts

```diff
@@ -262,4 +262,48 @@ export class HasBlockSuite extends RenderTest {
     this.assertComponent('', { 'data-has-block': 'false' });
     this.assertStableRerender();
   }
+
+  @test({ kind: 'glimmer' })
+  'has-block works within a yielded curried component invoked as variable'() {
+    this.registerComponent(
+      'Glimmer',
+      'ComponentWithHasBlock',
+      `<div data-has-block="{{has-block}}" ...attributes></div>`
+    );
+
+    this.registerComponent('Glimmer', 'Yielder', `{{yield (component 'ComponentWithHasBlock')}}`);
+
+    this.registerComponent(
+      'Glimmer',
+      'TestComponent',
+      `<Yielder as |componentWithHasBlock|>{{componentWithHasBlock}}</Yielder>`
+    );
+
+    this.render(`<TestComponent />`);
+
+    this.assertComponent('', { 'data-has-block': 'false' });
+    this.assertStableRerender();
+  }
+
+  @test({ kind: 'glimmer' })
+  'has-block works within a yielded curried component invoked using standard syntax'() {
```

Comment:
What is "standard syntax"?!

```suggestion
  'has-block works within a yielded curried component invoked via self-closing angle bracket invocation'() {
```